app.js

```diff
@@ -0,0 +1,353 @@
+/*global alert ,confirm ,window ,setInterval ,setTimeout ,document ,clearInterval*/
+    //  Get user info
+(function getUserInfo() {
+    'use strict';
+    var url = window.location.href,
+        userName,
+        userAge,
+        greetDate,
+        userNamePlace,
+        getTimeNow;
+
+    //  if game page has no data(string) in url, keep moving.
+    //  Otherwise, declare user's name and age variables.
+    if ("string" === typeof url.split("?")[1]) {
```

Comment:
This is a workaround, correct check should be, in English: "Does this string have a `?` characters ?", which translates into:
`-1 !== url.indexOf( '?' )` ( Google "JS indexOf" )